packages/foam-vscode/src/services/variable-resolver.ts

```diff
@@ -88,28 +93,44 @@ export class Resolver {
    */
   async resolveText(text: string): Promise<[Map<string, string>, string]> {
     const variablesInTemplate = findFoamVariables(text.toString());
-    const variables = variablesInTemplate.concat(
-      ...this.extraVariablesToResolve
+
+    const uniqVariableNamesInTemplate = new Set(
+      variablesInTemplate.map(variable => variable.name)
     );
-    const uniqVariables = [...new Set(variables)];
 
-    const resolvedValues = await this.resolveAll(uniqVariables);
+    const variables = variablesInTemplate.concat(
+      [...this.extraVariablesToResolve]
+        .filter(name => !uniqVariableNamesInTemplate.has(name))
```

Comment:
should this filtering be applied to the concat'ed array?

Reply:
I don't think so?

`uniqVariableNamesInTemplate` is derived from `variablesInTemplate`, so the filtering would do nothing?